src/include/migraphx/concat_opt.hpp

```diff
@@ -233,7 +233,7 @@ struct concat_optimization
     private_detail_te_handle_base_type& private_detail_te_get_handle()
     {
         assert(private_detail_te_handle_mem_var != nullptr);
-        if(not private_detail_te_handle_mem_var.unique())
+        if(private_detail_te_handle_mem_var.use_count() != 1)
```

Comment:
Same as above, should be use_count() > 1

Reply:
done